### ISSUE TITLE
BelongsTo field validation skipped when field in readonly state

### ISSUE THREAD

**Issue by rubenflush** [laravel/nova-issues#3671]
- Laravel Version: 7.30.4
- Nova Version: 3.29.0
- PHP Version: 8.0.13
- Database Driver & Version: MariaDB 10.3.32
- Operating System and Version: Debian buster

### Description:

When a `BelongsTo` field is in readonly state, the value can be manipulated using the `viaResource` post fields and validation will be skipped. This results in an invalid database record.

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create a resource: `User` with a `HasMany` field to the `Post` resource
2. Create a resource: `Post` with a `BelongsTo` field to the `User` resource which is readonly
3. Visit a User resource detail and click the 'Create post` button
4. Change the URL parameter value `viaResourceId` to 2 and reload the page (final URL http://nova-test.local/resources/posts/new?viaResource=users&viaResourceId=2&viaRelationship=posts)
5. The user field shows the first available value, but the value will not be posted because it is in readonly state. Instead, the query parameter is used while processing the data and not validated to the `BelongsTo` field values.
6. Store the new post. 

#### Users table migration
```php
<?php

declare(strict_types=1);

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

final class CreateUsersTable extends Migration
{
    public function up(): void
    {
        Schema::create('users', function (Blueprint $table): void {
            $table->increments('id');
            $table->string('name');
            $table->timestamps();
        });

        \Illuminate\Support\Facades\DB::table('users')
            ->insert([
                ['name' => 'First user (has access)'],
                ['name' => 'Second user (no access)'],
            ]);
    }

    public function down(): void
    {
        Schema::dropIfExists('users');
    }
}
```

#### Posts table migration
```
<?php

declare(strict_types=1);

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

final class CreatePostsTable extends Migration
{
    public function up(): void
    {
        Schema::create('posts', function (Blueprint $table): void {
            $table->increments('id');
            $table->integer('user_id');
            $table->string('title');
            $table->timestamps();
        });
    }

    public function down(): void
    {
        Schema::dropIfExists('posts');
    }
}
```

#### User entity
```php
<?php

declare(strict_types=1);

namespace App\Entities;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;

final class User extends Model
{
    protected static function booted()
    {
        self::addGlobalScope(function (Builder $builder): void {
            $builder->where('id', 1);
        });
    }

    public function posts()
    {
        return $this->hasMany(Post::class);
    }
}

```

#### Post entity
```php
<?php

declare(strict_types=1);

namespace App\Entities;

use Illuminate\Database\Eloquent\Model;

final class Post extends Model
{
    protected $fillable = ['user_id'];

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

```

#### User resource
```php
<?php

declare(strict_types=1);

namespace App\Nova;

use Illuminate\Http\Request;
use Laravel\Nova\Fields\HasMany;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Resource;
use App\Entities\User as UserEntity;

final class User extends Resource
{
    public static $model = UserEntity::class;

    public function fields(Request $request)
    {
        return [
            new Text('name'),
            new HasMany('Posts', 'posts', Post::class),
        ];
    }
}
```

#### Post resource

```php
<?php

declare(strict_types=1);

namespace App\Nova;

use Illuminate\Http\Request;
use Laravel\Nova\Fields\BelongsTo;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Resource;
use App\Entities\Post as PostEntity;

final class Post extends Resource
{
    public static $model = PostEntity::class;

    public function fields(Request $request)
    {
        return [
            (new BelongsTo('user', 'user', User::class))->readonly(null !== $request->viaResource),
            new Text('title'),
        ];
    }
}

```

**Comment by crynobone** [comment]
Please provide reproducing code to explain the issue.

**Comment by rubenflush** [comment]
Reproducing code included in description

**Comment by rubenflush** [comment]
Any update on this as the issue is closed without comment?

**Comment by crynobone** [comment]
> Any update on this as the issue is closed without comment?

Any issue assigned and label as `bug` will only be closed after a PR has been merged. However, we still waiting for the patch release before updating the ticket.

**Comment by crynobone** [comment]
Fixed in v3.31.0: https://nova.laravel.com/releases/3.31.0
